### PR TITLE
Global hotkey setup

### LINKED ISSUE #5: get input for keybinds (pynput)
should be easy enough to find out how to do: google "how to make a keylogger in python"

but basically we won't store any of the keystrokes, just sending signals to our tracker to do things when receiving certain keystrokes (for example, to quit)

later we should make the keybinds configurable, but that would come after we add more keybinds

### PR DESCRIPTION
**Pull request by AitanSingh**
Resolves #5

### CODE REVIEW

File: src/keylogger.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from pynput import keyboard
+
+pressed_keys = set()
+def keyPressed(key):
+    char = key.char
+    pressed_keys.add(char)
+    print(f"Key pressed: {key}")
```
Comment (jahndan) [inline src/keylogger.py]: not the biggest deal since we're not going to keep printing it, but handle the exception. it's literally [an example given in the documentation](https://pynput.readthedocs.io/en/latest/keyboard.html#monitoring-the-keyboard) -- especially given that **the keyboard listener stops itself if an exception is raised in the callback**

also the documentation suggests `keyboard.Listener` is already doing `threading.Thread` stuff so you may want to bring back parts of the threaded code you initially wrote

File: src/keylogger.py
Comment (jahndan) [inline src/keylogger.py:1]: also I know I said to look up how to make a keylogger, but I didn't mean to literally call it that
we're using it to establish [tracker keybinds](https://pynput.readthedocs.io/en/latest/keyboard.html#monitoring-the-keyboard)—strictly non-malicious purposes

^^ the documentation I'm linking here has example code explicitly for this use-case

File: src/keylogger.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from pynput import keyboard
+
+pressed_keys = set()
```
Comment (jahndan) [inline src/keylogger.py]: for the record, if you're bringing back the explicitly threaded stuff, you should use a queue and enqueue key press events for the main tracker process to handle and dequeue those events
-- event handling doesn't need to happen immediately, and more importantly pynput documentation suggests that the callback functions themselves should be quick to return

File: src/main.py [hunk 3]
```diff
@@ -56,3 +59,4 @@ def print_arg_help():
             if (opencv.waitKey(1) & 0xFF) == ord("q"):
                 opencv.destroyAllWindows()
                 break
+        listener.join()
```
Comment (jahndan) [inline src/main.py]: does this need to be here at all? the script terminates after it breaks out of the while loop
-- for emphasis, what we're (going to be) doing is capturing key presses in a non-blocking fashion (see the last bit in [this example code](https://pynput.readthedocs.io/en/latest/keyboard.html#monitoring-the-keyboard))
-- if we want to reintroduce threading, it may be different entirely

**Review by jahndan — CHANGES_REQUESTED** [review]
TL;DR: read the pynput documentation and try to set this up to run without interfering/slowing down the frame processing

**Review by jahndan — APPROVED** [review]
Good. Wait on backend merge for this PR.

**Comment by jahndan** [comment]
why does github let me resolve conflicts if it's only going to do a merge commit ... and merge commits aren't allowed in the repo

**Comment by jahndan** [comment]
gotta rebase locally and push i guess
